### PR TITLE
chore: improve Upstream/CodeMirror

### PR DESCRIPTION
**Pull request by juzhiyuan**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with a detailed description.

- [x] use UpperCase for JSON/YAML selector in CodeMirror
- [x] added more description for Upstream List
- [x] added more proper fields description for Upstream Form (Chinese: Done; English: **TODO**)
- [x] added retries field for health check
- [ ] added default values for health check

**Related issues**

None

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
as for default values, we need to refer to https://github.com/api7/lua-resty-healthcheck/tree/master/docs , I will file a new PR to do that.

**Comment by juzhiyuan** [comment]
@LiteSun 

image

I have some doubts:

1. Why FE doesn't have coverage 😳
2. I don't change backend files, but this PR reduces some coverage.

**Comment by juzhiyuan** [comment]
@iamayushdas @KishaniKandasamy @stu01509 You could also review this PR :)

**Comment by stu01509** [comment]
> @iamayushdas @KishaniKandasamy @stu01509 You could also review this PR :)

Sure 😃

**Review by iamayushdas — APPROVED** [review]
👍🏻

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 1]
```diff
@@ -257,15 +249,16 @@ const UpstreamForm: React.FC<Props> = forwardRef(
 
     const ActiveHealthCheck = () => (
       <>
-        <Form.Item label={formatMessage({ id: 'page.upstream.step.healthyCheck.active.timeout' })}>
+        <Form.Item label={formatMessage({ id: 'page.upstream.step.healthyCheck.active.timeout' })} tooltip="主动健康检查的套接字的超时时间">
           <Form.Item name={['checks', 'active', 'timeout']} noStyle>
```
Comment (stu01509) [inline web/src/components/Upstream/UpstreamForm.tsx:253]: `主动健康检查的套接字的超时时间` maybe need to declare in locales file?

Comment (juzhiyuan) [reply web/src/components/Upstream/UpstreamForm.tsx:253]: oh yes!

File: web/src/components/RawDataEditor/locales/en-US.ts [hunk 2]
```diff
@@ -16,5 +16,5 @@
  */
 export default {
   'component.rawDataEditor.tip': "Don't support edit mode currently",
-  'component.rawDataEditor.title': 'Raw Data Editor',
+  'component.rawDataEditor.title': 'CodeMirror',
```
Comment (liuxiran) [inline web/src/components/RawDataEditor/locales/en-US.ts:19]: May I ask why did not use `Raw Data Editor` here?  It seems that this is  easier to understand

Comment (juzhiyuan) [reply web/src/components/RawDataEditor/locales/en-US.ts:19]: 😂 but we don't have a uniform  name for now, `Create with Editor`/`Raw Data Editor`

File: web/src/locales/en-US/component.ts [hunk 3]
```diff
@@ -25,12 +25,12 @@ export default {
   'component.global.document': 'Document',
   'component.global.enable': 'Enable',
   'component.global.scope': 'Scope',
-  'component.global.data.editor': 'Data Editor',
+  'component.global.data.editor': 'CodeMirror',
   'component.global.delete': 'Delete',
   'component.global.cancel': 'Cancel',
   'component.global.submit': 'Submit',
   'component.global.create': 'Create',
-  'component.global.createWithEditor': 'Create with Editor',
```
Comment (LiteSun) [inline web/src/locales/en-US/component.ts:33]: I think `Create with Editor` is easier to understand, those who have not used `CodeMirror ` before may not know what codemirror does. 🤔

**Comment by KishaniKandasamy** [comment]
> @iamayushdas @KishaniKandasamy @stu01509 You could also review this PR :)




👍🏻

File: web/src/components/RawDataEditor/locales/zh-CN.ts [hunk 4]
```diff
@@ -16,5 +16,5 @@
  */
 export default {
   'component.rawDataEditor.tip': '目前暂不支持编辑',
-  'component.rawDataEditor.title': '元数据编辑器',
+  'component.rawDataEditor.title': 'CodeMirror',
```
Comment (KishaniKandasamy) [inline web/src/components/RawDataEditor/locales/zh-CN.ts:19]: English in  zh-CN.ts🤔

Comment (iamayushdas) [reply web/src/components/RawDataEditor/locales/zh-CN.ts:19]: I guess there is no suitable word for that in cn

Comment (juzhiyuan) [reply web/src/components/RawDataEditor/locales/zh-CN.ts:19]: Yes.

BTW, I want to make sure most of users know what CodeMirror is 😂

Does this word make sense to you? :)

Comment (juzhiyuan) [reply web/src/components/RawDataEditor/locales/zh-CN.ts:19]: Maybe I need add some descriptions for CodeMirror and  still keeps using this word? 😊

File: web/src/locales/en-US/component.ts (hunk 3 above)
Comment (juzhiyuan) [reply web/src/locales/en-US/component.ts:33]: Not sure, Create with Editor is more like a description than a feature name IMO :)

How about other changes?

**Comment by juzhiyuan** [comment]
Hi, I noticed we have comments all about CodeMirror, how about adding some descriptions for this component? I don't have a better name yet, I would still prefer keeping using CodeMirror than Create with Editor. What's your opinion? 

BTW, how about other changes in this PR? Do they make sense to you?

**Comment by juzhiyuan** [comment]
Oops! When I want to select more reviewers on GitHub app, it unselect some of reviewers.

File: web/src/components/RawDataEditor/locales/zh-CN.ts (hunk 4 above)
Comment (iamayushdas) [reply web/src/components/RawDataEditor/locales/zh-CN.ts:19]: That would be good for everyone then, who are having problem with this word right now.

File: web/src/components/RawDataEditor/locales/en-US.ts (hunk 2 above)
Comment (liuxiran) [reply web/src/components/RawDataEditor/locales/en-US.ts:19]: How about looking at this issue from another angle:
`Create with Editor` is an action that can arouse `CodeMirror`, and `Raw Data Editor` can be considered as a name we gave to `CodeMirror`. So we just need to unify the action name and the `CodeMirror` name.

for the `CodeMirror` name, `Raw Data Editor `is a better choice IMHO

File: web/src/components/RawDataEditor/locales/zh-CN.ts (hunk 4 above)
Comment (KishaniKandasamy) [reply web/src/components/RawDataEditor/locales/zh-CN.ts:19]: > Yes.
> 
> BTW, I want to make sure most of users know what CodeMirror is 😂
> 
> Does this word make sense to you? :)

yes :)

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 5]
```diff
@@ -771,7 +775,12 @@ const UpstreamForm: React.FC<Props> = forwardRef(
                 {
                   label: formatMessage({ id: 'page.upstream.step.healthyCheck.active' }),
                   name: ['checks', 'active'],
-                  component: <ActiveHealthCheck />,
+                  component: (
+                    <>
+                      <ActiveHealthCheck />
+                      <Divider orientation="left" plain></Divider>
```
Comment (qian0817) [inline web/src/components/Upstream/UpstreamForm.tsx:781]: collapse empty tag
```suggestion
                      <Divider orientation="left" plain/>
```

File: web/src/pages/Upstream/List.tsx [hunk 6]
```diff
@@ -136,12 +136,11 @@ const Page: React.FC = () => {
             <PlusOutlined />
             {formatMessage({ id: 'component.global.create' })}
           </Button>,
-          <Button type="primary" onClick={() => {
+          <Button type="default" onClick={() => {
```
Comment (qian0817) [inline web/src/pages/Upstream/List.tsx:139]: I think the original way looks better.
At the same time, If really need to do this, also need to modify the style of other pages.

File: web/src/components/Upstream/UpstreamForm.tsx (hunk 5 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/UpstreamForm.tsx:781]: Oh yes, and we'd better add a space before / :)

File: web/src/pages/Upstream/List.tsx (hunk 6 above)
Comment (juzhiyuan) [reply web/src/pages/Upstream/List.tsx:139]: not sure if we really need the primary button here 🤔

**Review by bzp2010 — APPROVED** [review]

**Review by Yiyiyimu — APPROVED** [review]

**Review by KishaniKandasamy — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Pull request merged**